server/vulnerabilities/nvd/cve_test.go

```diff
@@ -230,52 +154,155 @@ func TestTranslateCPEToCVE(t *testing.T) {
+				"CVE-2016-7642",
+				"CVE-2016-7645",
+				"CVE-2016-7646",
+				"CVE-2016-7648",
+				"CVE-2016-7649",
+				"CVE-2016-7652",
+				"CVE-2016-7654",
+				"CVE-2016-7656",
+				"CVE-2017-2383",
+			},
+			continuestoUpdate: true,
+		},
+		"cpe:2.3:a:clickstudios:passwordstate:9.5.8.4:*:*:*:*:chrome:*:*": {
+			includedCVEs:      []string{"CVE-2022-4610", "CVE-2022-4611", "CVE-2022-4613", "CVE-2022-4612"},
+			continuestoUpdate: true,
+		},
+		"cpe:2.3:a:avira:password_manager:2.18.4.38471:*:*:*:*:firefox:*:*": {
+			includedCVEs:      []string{"CVE-2022-28795"},
+			continuestoUpdate: true,
+		},
+		"cpe:2.3:a:zoom:zoom:5.0.4301.0407:*:*:*:*:chrome:*:*": {
+			excludedCVEs:      []string{"CVE-2021-28133"}, // CVE-2021-28133 is a vulnerability in the Zoom application, not the extension.
+			continuestoUpdate: true,
+		},
+		"cpe:2.3:a:bitwarden:bitwarden:1.55.0:*:*:*:*:firefox:*:*": {
+			excludedCVEs:      []string{"CVE-2023-38840"}, // CVE-2023-38840 is a vulnerability in the Bitwarden application, not the extension.
+			continuestoUpdate: true,
+		},
+	}
+
+	t.Run("find_vulns_on_cpes", func(t *testing.T) {
+		t.Parallel()
 
-			cveLock := &sync.Mutex{}
-			var cvesFound []string
-			ds.InsertSoftwareVulnerabilityFunc = func(ctx context.Context, vuln fleet.SoftwareVulnerability, src fleet.VulnerabilitySource) (bool, error) {
-				cveLock.Lock()
-				defer cveLock.Unlock()
-				cvesFound = append(cvesFound, vuln.CVE)
-				return false, nil
+		ds := new(mock.Store)
+
+		softwareIDToCPEs := make(map[uint]string)
+		ds.ListSoftwareCPEsFunc = func(ctx context.Context) ([]fleet.SoftwareCPE, error) {
+			var softwareCPEs []fleet.SoftwareCPE
+			i := uint(0)
+			for cpe := range cveTests {
+				softwareCPEs = append(softwareCPEs, fleet.SoftwareCPE{CPE: cpe, SoftwareID: i})
+				softwareIDToCPEs[i] = cpe
+				i++
 			}
-			ds.DeleteOutOfDateVulnerabilitiesFunc = func(ctx context.Context, source fleet.VulnerabilitySource, duration time.Duration) error {
-				return nil
+			return softwareCPEs, nil
+		}
+
+		cveLock := &sync.Mutex{}
+		cvesFound := make(map[string][]string)
+		ds.InsertSoftwareVulnerabilityFunc = func(ctx context.Context, vuln fleet.SoftwareVulnerability, src fleet.VulnerabilitySource) (bool, error) {
+			cveLock.Lock()
+			defer cveLock.Unlock()
+
+			cpe, ok := softwareIDToCPEs[vuln.SoftwareID]
+			if !ok {
+				return false, fmt.Errorf("software id -> cpe not found: %d", vuln.SoftwareID)
 			}
+			cvesFound[cpe] = append(cvesFound[cpe], vuln.CVE)
+			return false, nil
+		}
+		ds.DeleteOutOfDateVulnerabilitiesFunc = func(ctx context.Context, source fleet.VulnerabilitySource, duration time.Duration) error {
+			return nil
+		}
 
-			_, err := TranslateCPEToCVE(ctx, ds, tempDir, kitlog.NewLogfmtLogger(os.Stdout), false, 1*time.Hour)
-			require.NoError(t, err)
+		_, err := TranslateCPEToCVE(ctx, ds, tempDir, kitlog.NewLogfmtLogger(os.Stdout), false, 1*time.Hour)
+		require.NoError(t, err)
 
-			printMemUsage()
+		printMemUsage()
```

Comment:
Is this needed? If needed, I think `t.Log(...)` should be used. Reference: https://pkg.go.dev/testing#B.Log

Reply:
Don't think so. Let's remove, it's not functional.
[Fix PR comments](https://github.com/fleetdm/fleet/pull/14791/commits/7a97762a299cd7d537ac14f95d5c0db733c6c972)